jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/UserControllerTest.groovy

```diff
@@ -44,153 +37,118 @@ import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.
 @TransactionConfiguration(transactionManager = "transactionManager", defaultRollback = true)
 @Transactional
 class UserControllerTest extends Specification {
-    @Autowired ComponentService componentService
     @Autowired Users users
     @Autowired GroupsService groupsService
 
-    @Autowired MockMvc mockMvc
-
-    private Component forum
-    private User adminUser;
-
     def setup() {
         groupsService.create()
-        forum = componentService.createForumComponent()
-
-        adminUser = new User()
-        users.created(adminUser).withPermissionOn(forum, GeneralPermission.ADMIN)
     }
 
-    def 'test show user groups'() {
+    def 'test must return all the groups of a user'() {
         given: 'Admin login'
-            def session = users.signIn(adminUser)
+            def session = users.signInAsAdmin()
             def expectedGroups = [AdministrationGroup.BANNED_USER.getName(), AdministrationGroup.USER.getName()]
         and: 'User created with groups BANNED_USER and USER'
             def user = new User()
             users.createdWithoutAccess(user).withGroups(expectedGroups);
         when: 'User fetch groups'
-            def result = mockMvc.perform(get('/user/' + users.userIdByUsername(user.username) + '/groups')
-                .session(session as MockHttpSession)
-                .contentType(MediaType.APPLICATION_JSON)
-            )
-        then: 'Fetched groups list'
-            result.andExpect(status().isOk())
-                .andExpect(jsonPath('$.status', is('SUCCESS')))
-                .andExpect(jsonPath('$.result', hasItems(groupsService.getIDsByName(expectedGroups) as int[])))
+            List<Long> receivedGroups = users.fetchGroups(session, user);
+        then: 'Fetched all user groups'
+            assertThat(receivedGroups, is(groupsService.getIdsByName(expectedGroups)))
     }
 
-    def 'try fetch user groups without access'() {
+    def 'test without access must return fail result'() {
         given: 'User created without access and login'
             def user = new User()
             users.createdWithoutAccess(user);
             def session = users.signIn(user)
         when: 'User fetch groups without access'
-            def result = mockMvc.perform(get('/user/' + users.userIdByUsername(user.username) + '/groups')
-                .session(session as MockHttpSession)
-                .contentType(MediaType.APPLICATION_JSON)
-            )
+            users.fetchGroups(session, user);
         then:
-            result.andExpect(status().isOk())
-                .andExpect(jsonPath('$.status', is('FAIL')))
+            thrown(WrongResponseException)
     }
 
-    def 'try fetch user groups without authorization'() {
+    def 'test without authorization should be redirection'() {
```

Comment:
`user must be redirected to login page if he tries to access groups without authentication`